File: Monstone_Android_Client/app/src/main/java/com/example/nightwingky/monstone/user_action/user_login/UserLoginHttp.java
package com.example.nightwingky.monstone.user_action.user_login;

import java.io.IOException;

import okhttp3.FormBody;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by nightwingky on 17-3-12.
 */

public class UserLoginHttp {

    public static String userLoginQuery(String URL, String username, String pwd) throws IOException {
        OkHttpClient okHttpClient = new OkHttpClient();

        FormBody formBody = new FormBody.Builder()
                .add("loginName", username)
                .add("password", pwd)
                .build();

        Request request = new Request.Builder()
                .url(URL)
                .post(formBody)
                .build();

        Response response = okHttpClient.newCall(request).execute();

        if(response.isSuccessful()) {
            return response.body().string();
        } else {
            return null;
        }
    }
}

File: kotlin_client/app/src/main/java/com/graduation/nightwingky/book_drifting/monstone/user_action/user_login/UserLoginHttp.kt
package com.graduation.nightwingky.book_drifting.monstone.user_action.user_login

import java.io.IOException

import okhttp3.FormBody
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response

/**
 * Created by nightwingky on 17-3-12.
 */

object UserLoginHttp {

    @Throws(IOException::class)
    fun userLoginQuery(URL: String, username: String, pwd: String): String? {
        val okHttpClient = OkHttpClient()

        val formBody = FormBody.Builder()
                .add("loginName", username)
                .add("password", pwd)
                .build()

        val request = Request.Builder()
                .url(URL)
                .post(formBody)
                .build()

        val response = okHttpClient.newCall(request).execute()

        return if (response.isSuccessful) {
            response.body()!!.string()
        } else {
            null
        }
    }
}
